examples/mount.rs

```diff
@@ -0,0 +1,18 @@
+#[macro_use] extern crate nickel;
+
+use nickel::{Nickel, Mountable, StaticFilesHandler};
+
+fn main() {
+    let mut server = Nickel::new();
+
+    /*
+     * Fall-through behaviour, if StaticFilesHandler does not find a matching file,
+     * the request uri must be reset so that it can be matched against other middleware.
+     */
+    server.mount("/static/files/", StaticFilesHandler::new("examples/assets/"));
```

Comment:
Looking at this again, the rewrite rules would make this `"/static/files"`? In that case, does this mean that the mount will activate for paths such as `"/static/files_and_stuff/bar"` and pass the uri as `"_and_stuff/bar"`?

I'm also curious about the behavior of `"/"`, as the uri passed to the mounted middleware would be `""`  if the base route is caught (i.e. the client sends `"/"`)? I would have expected `"/"` to be passed?

I'm not really sure what motivates the rewrite rules, can you add a comment to `rewrite_mount_point` with expected behavior?